docs/content/developer/workshops/core/I-introduction-to-smart-contracts.mdx

```diff
@@ -0,0 +1,361 @@
+    public struct Grant has key {
+        id: UID,
+        student: address,
+        amount: u64,
+    }
+
+    /// Emitted when a grant is minted.
+    public struct GrantMinted has copy, drop, store {
+        student: address,
+        amount: u64,
+        grant_id: ID,
+    }
+}
+```
+
+Notice how both `AdminCap` and `Grant` have the `key` ability. Instantiating either will create an object with a `UID` you can later transfer.
+
+:::info What are UID and ID?
+- `UID` is the unique token stored inside the object that gives it identity; it isn’t copyable and lives only within the object.
+- `ID` is the canonical identifier value derived from a reference to the object (`object::id(&obj)`), great for events and lookups.
+- Think: `UID` is the sealed certificate inside the box; `ID` is the serial number on the receipt.
+- Learn more: [UID and ID](../../iota-101/objects/uid-id.mdx)
+:::
+
+## Give someone the keys (module initializer)
+
+Now, let’s make sure the package publisher receives the admin capability when the package is published.
+
+Append this function inside the same module:
+
+```move
+    /// Runs once at package publish. Transfers AdminCap to the publisher.
+    fun init(ctx: &mut TxContext) {
+        let cap = AdminCap { id: object::new(ctx) };
+        let publisher = tx_context::sender(ctx);
+        transfer::transfer(cap, publisher);
+    }
+```
+
+:::info What are init and TxContext?
+- `fun init` runs once at publish time to bootstrap state (e.g., mint an `AdminCap`). It cannot be called again.
+- `TxContext` is the per‑transaction context: it gives you the sender and lets you create fresh object IDs (`object::new(ctx)`).
+- Like a deployment migration that seeds initial rows and grants the first admin.
+- Learn more: [Module Initializers](../../iota-101/move-overview/init.mdx)
+:::
+
+Quick checkpoint:
+
+```bash
+iota move build
+```
+
+If you hit snags, see [Build, Test, Debug](../../getting-started/build-test.mdx) and the [Move CLI](../../references/cli/move.mdx).
+
+## The mint you’ll write
+
+Here’s the story your `mint` function should tell when it runs:
+
+- Create a fresh `Grant` object with the intended `student` and an `amount`.
+- Emit a `GrantMinted` event that includes the new object’s `ID`.
+- Transfer the newly created `Grant` to the `student` address.
+
+We’ll leave the full body for you to implement. Start with the signature and a placeholder, then iterate.
+
+Append this inside the same module:
+
+```move
+    public entry fun mint(
+        student: address,
+        amount: u64,
+        _cap: &AdminCap,
+        ctx: &mut TxContext
+    ) {
+        // TODO: Create Grant, emit event, transfer to `student`.
+        // Hints: object::new(ctx), object::id(&grant), transfer::transfer(grant, student)
+        abort 0;
+    }
+```
+
+:::info What are entry functions and transfers?
+- `public entry` functions are the only functions callable as transaction entry points; think of them as your public API.
```

Comment:
But isn't any function a "public API" as long as it is at least public? ;)
Also do we really need a public entry function here? If we do, this should mention why exactly